subsys/fs/fs.c

```diff
@@ -22,10 +22,10 @@
 LOG_MODULE_REGISTER(fs);
 
 /* list of mounted file systems */
-static sys_dlist_t fs_mnt_list;
+static sys_dlist_t fs_mnt_list = SYS_DLIST_STATIC_INIT(&fs_mnt_list);
 
 /* lock to protect mount list operations */
-static struct k_mutex mutex;
+K_MUTEX_DEFINE(mutex);
```

Comment:
Keep the mutex instance `static`:

```suggestion
static K_MUTEX_DEFINE(mutex);
```